src/components/Photo/Selfie.js

```diff
@@ -11,29 +11,92 @@ import style from './style.css'
 
 type State = {
   hasBecomeInactive: boolean,
+  snapshotBuffer: Array<{
+    blob: Blob,
+    base64: string
+  }>,
 }
 
 type Props = {
   onCapture: Function,
   renderFallback: Function,
   trackScreen: Function,
-  inactiveError: Object
+  inactiveError: Object,
+  useMultipleSelfieCapture: boolean,
+  snapshotInterval: number,
 }
 
 export default class Selfie extends Component<Props, State> {
   webcam = null
+  snapshotIntervalRef: ?IntervalID = null
 
   state: State = {
     hasBecomeInactive: false,
+    snapshotBuffer: [],
   }
 
   handleTimeout = () => this.setState({ hasBecomeInactive: true })
 
-  onScreenshotCapture = (blob: Blob, base64: string, sdkMetadata: Object) => {
-    this.props.onCapture({ blob, base64, sdkMetadata })
+  buildCapture = (blob: Blob, base64: string, sdkMetadata: Object, name: string) => ({
+    blob: new File([blob], `${name}.png`, {
+      type: "image/png"
+    }),
+    base64,
+    sdkMetadata
+  })
+
+  snapshotData = () => ({
+    /* Attempt to get the 'ready' snapshot. But, if that fails, try to get the fresh snapshot - it's better
+       to have a snapshot, even if it's not an ideal one */
+    snapshot: this.state.snapshotBuffer[0] || this.state.snapshotBuffer[1],
+  })
+
+  handleSelfie = (blob: Blob, base64: string, sdkMetadata: Object) => {
+    const selfieCaptureData = this.buildCapture(blob, base64, sdkMetadata, 'applicant_selfie')
+    const snapshotData = this.snapshotData()
+    const captureData = this.props.useMultipleSelfieCapture ?
+      { ...snapshotData, ...selfieCaptureData } : selfieCaptureData
+    this.props.onCapture(captureData)
+  }
+
+  handleSnapshot = (blob: Blob, base64: string, sdkMetadata: Object) => {
+    /* When taking snapshots we want to ensure that the snapshot we provide is, where possible,
+      older than the interval. As we take a snapshot every interval, that means we need to
+      store the previous snapshot so that we can use that if the capture is taken too soon
+      after the latest snapshot (e.g. 500ms after the last snapshot on a 1000ms interval).
+
+      Therefore we use a simple two item collection as our buffer:
+
+      ['ready snapshot', 'fresh snapshot']
+
+      The first item is the older of the two snapshots, and will be used when capturing the selfie.
+      The second item is the 'fresh' slot which will not be suitable until the next interval tick at which
```

Comment:
I'm a little confused about this set-up. Why is it that the "fresh" slot "isn't suitable" until the next interval tick?
`this.buildCapture` looks synchronous to me, I don't get why waiting a tick should accomplish anything?

Reply:
I've updated the comment